app/controllers/user/UserProfile.controller.ts

```diff
@@ -5,6 +5,62 @@ import UserProfile from '../../models/UserProfile';
 import { IProfileRequest } from '../../request/IProfileRequest';
 import { IUserRequest } from '../../request/IRequest';
 
+/**
+ * @api {get} NEED ENDPOINT URL Shows all of the profile information for a certain username.
+ * @apiName ShowProfileforUsername
```

Comment:
`Show` can probably be replaced to `Get`, and `Username` to `User` (for more generic reading) since all users endpoints are based on the id of the user and not the username.